docs/design/proposals/native-binding/native-binding.md

```diff
@@ -0,0 +1,245 @@
+
+ACK users often need to access fields from the infrastructure created using K8s
+custom resources into their deployment specifications. One example of this would
+be to take the [RDS DBCluster
+`Status.Endpoint`](https://aws-controllers-k8s.github.io/community/reference/rds/v1alpha1/dbcluster/#status)
+and inject it as an environment variable into a web application that requires a
+database. Currently, users can only achieve this with two separate steps.
+Firstly, the user must create the ACK resource and wait until it reconciles.
+Then they must manually read the status from the resource and inject it into the
+manifest of their deployment. Given this process requires a manual step, and
+specific order of operations, it is not compatible with GitOps systems to create
+infrastructure in the same manifests as an application.
+
+This design aims to address two separate use cases:
+
+1. An application team manages their own ACK resources and K8s deployments. They
+   wish to dynamically link the status of an ACK resource to the environment
+   variables of their deployment.
+2. An infrastructure team manages the ACK resources on behalf of application
+   teams. The application teams do not have read/write access to the ACK
+   resource directly, but require access to the fields from them in order to
+   deploy their application.
+
+## Requirements
+
+1. Allow the cluster operator to configure an ACK controller to write the value
+  of a custom resource Spec or Status field to either a ConfigMap or a Secret
+2. Export to any namespace (for cluster-scoped controllers only)
+3. The exported value should stay in sync as the resource changes
+4. Resources should not need to be updated to add/remove bindings
+
+## Existing Solutions
+
+### Redhat Service Binding Operator
+
+>“[Redhat Service Binding Operator] enables developers to connect their
+>application to backing services with a consistent and predictable experience“ -
+>https://github.com/redhat-developer/service-binding-operator
+
+
+The [Redhat Service Binding
+Operator](https://github.com/redhat-developer/service-binding-operator) (herein
+called the SBO) uses annotations on resources to import field values from other
+resources. The main purpose of SBO is to standardise the system for injecting
+secret information from database services into application deployment
+environment variables. The SBO can automatically detect and bind to services
+only from a limited number of [supported
+operators](https://github.com/redhat-developer/service-binding-operator#known-bindable-operators). 
+
+SBO is not suitable for this design as it can only support a limited number of
+fields based on pre-configured database bindings. That is, it can handle the
+case where an application developer wants to inject a PostgreSQL database
+connection string into a deployment, but little more than that. SBO does not
+support exporting to configmap or secrets directly.
```

Comment:
> SBO does not support exporting to configmap or secrets directly.

I might be reading their documentation backwards or not understanding it fully, but I feel it supports configmaps and secrets. Am I reading the [documentation](https://redhat-developer.github.io/service-binding-operator/userguide/exposing-binding-data/olm-descriptors.html) incorrectly?